### PR TITLE
Normalize percent-encoded bytes before comparison

### PR DESCRIPTION
**Pull request by sethmlarson**
urllib3 replaced it's URL parser with `rfc3986` for more compliant parsing (urllib3/urllib3#1487). We've also got an effort underway to run downstream tests before releases (urllib3/urllib3#1508) and during that process we've discovered that requests requires percent-encoded bytes to keep their casing.

Adding this to the unit tests would allow either version of pre- and post-rfc3986 urllib3 to pass requests unit tests. I don't know if the `Request.url` attribute needs to maintain it's casing in this situation or not, if it does this is not the solution. Just putting this together as a potential solution.

### CODE REVIEW

**Comment by sigmavirus24** [comment]
Closing and re-opening to trigger a new build

**Comment by sethmlarson** [comment]
Is a newer pytest version not compatible with one of our plugins / our test setup somehow?

**Comment by sigmavirus24** [comment]
Beats me

**Comment by sethmlarson** [comment]
I guess I'll have to poke around with this more and see why we're getting these funky errors on test load.

**Comment by sethmlarson** [comment]
Just as I suspected, pytest 4.1 is the problem. Had to deal with this in multiple places at work as well. :( Now to figure out why [Python 3.6 is still installing pytest 4.1.1 despite being locked to `<4.1`](https://travis-ci.org/requests/requests/jobs/481793727#L499).

...and it's installing on Python 3.7 on AppVeyor? Is this a bug with Pipenv and `--skip-lock`?

**Comment by sethmlarson** [comment]
I don't know what you think about this @sigmavirus24 / @kennethreitz, I don't want to remove the `--skip-lock` because we're getting really outdated dependencies but it looks like `pipenv` is having trouble with the version specifier?

File: .travis.yml [hunk 1]
```diff
@@ -38,5 +38,5 @@ jobs:
       dist: xenial
       sudo: true     
     - stage: coverage
-      python: 3.6
+      python: '3.6'
```
Comment (sigmavirus24) [inline .travis.yml:41]: This is odd

**Comment by sethmlarson** [comment]
Thanks for merging this!

**Pull request merged**